src/analysis/types.js

```diff
@@ -2,17 +2,70 @@
 
 import {type NodeAddressT, type EdgeAddressT} from "../core/graph";
 
+/**
+ * Represents a "Type" of node in the graph.
+ *
+ * An node is considered a member of this type if
+ * its address has this type's prefix as an address prefix,
```

Comment:
While I think I can deduce what you're saying, the repetition of words makes the pieces of functionality you're referring to a bit obscure. Perhaps edit to say "A Node is considered a member of NodeAddressT if..." and include a link to `Address.hasPrefix` semantics (which, I believe, is here: https://github.com/sourcecred/sourcecred/blob/83fa29688ed97c996212fe8d02abe7d2a32ac4c1/src/core/graph.js#L299)

Reply:
Thanks for this comment. I realized that I can do a better job explaining how types work, and re-wrote the documentation. Please take another look and tell me what you think.

Reply:
New version is a really solid intro. 👍  Does a great job of presenting first principles in a highly readable form, then showing how they combine/build on each other.